bg_utils/__init__.py

```diff
@@ -404,36 +433,24 @@ def _ensure_roles():
                 'bg-system-create',
                 'bg-system-read',
                 'bg-system-update',
-            ]
-        ).save()
+            ],
+        ),
+    ]
 
-    try:
-        Role.objects.get(name='bg-admin')
-    except DoesNotExist:
-        logger.warning('Role bg-admin missing, about to create')
-        Role(
-            name='bg-admin',
-            description='Allows all actions',
-            permissions=['bg-all']
-        ).save()
+    # Only create convenience roles if this is a fresh database
+    if Role.objects.count() == 0:
+        logger.warning('No roles found: creating convenience roles')
 
-    try:
-        Role.objects.get(name='bg-anonymous')
-    except DoesNotExist:
-        logger.warning('Role bg-anonymous missing, about to create')
-        Role(
-            name='bg-anonymous',
-            description='Special role used for non-authenticated users',
-            permissions=[
-                'bg-command-read',
-                'bg-event-read',
-                'bg-instance-read',
-                'bg-job-read',
-                'bg-queue-read',
-                'bg-request-read',
-                'bg-system-read',
-            ]
-        ).save()
+        for role in convenience_roles:
+            try:
+                # Since we have a race potential here catch the case where
+                # another process has already created the role
+                _create_role(role)
+            except NotUniqueError:
+                logger.warning('Unable to create role %s' % role.name)
```

Comment:
This is a misleading logging message. It should probably say, 'Role %s already created.' % role.name